docs/software_arch.md

```diff
@@ -0,0 +1,75 @@
+are all defined in `deadline.maya_submitter.mel_commands`. These simply create a UI window to display to the user.
+
+Fundamentally, what this submitter is doing is creating a [Job Bundle](https://docs.aws.amazon.com/deadline-cloud/latest/developerguide/build-job-bundle.html)
+and using the GUI creation code in the [`deadline` Python package](https://pypi.org/project/deadline/) to generate the UI
+that is displayed to the user. The important parts to know about in a job bundle are:
+
+1. The job template file. The submitter code dynamically generates the template based on properties of the specific scene file
+   that is loaded. For example, it may contain a Step for each layer of the scene to render.
+   Note: All job template files are currently derived from a standard static job template located at
+   `src/deadline/maya_submitter/default_maya_job_template.yaml`.
+2. Asset references. These are the files that the job, when submitted, will require to be able to run. The submitter contains code
+   that introspects the scene that is loaded to automatically discover these, and the submitter UI allows the end-user to modify
+   the list of files.
+
+The job submission itself is handled by functionality within the `deadline` package that is hooked up when the UI is created.
+
+## Maya Adaptor Application
+
+See the [README](../README.md#maya-adaptor) for background on what purpose the adaptor application serves.
+
+The implementation of the adaptor for Maya has two parts:
+
+1. The adaptor application itself whose code is located in `src/deadline/maya_adaptor/MayaAdaptor`. This is the
+   implementation of the command-line application (named `maya-openjd`) that is run by Jobs created by the Maya submitter.
+2. A "MayaClient" application located in `src/deadline/maya_adaptor/MayaClient`. This is an application that is
+   run within Maya itself by the adaptor application when it launches Maya. The MayaClient facilitates communication
+   between the adaptor itself and the running Maya application; communication to tell Maya to, say, load a scene file,
+   or render frame 20 of the loaded scene.
+
+The adaptor application itself is built using the [Open Job Description Adaptor Runtime](https://github.com/OpenJobDescription/openjd-adaptor-runtime-for-python)
+package. This package supplies the application entrypoint that defines and parses the command-line subcommands and options, as well as
+the business logic that drives the state machine of the adaptor itself. Please see the README for the runtime package for information on
+the lifecycle states of an adaptor, and the command line options that are available. 
+
+Digging through the code for the adaptor, you will find that the `MayaAdaptor.on_start()` method is where the Maya application is started.
+Rather than starting it with command-line options that say to render a specific scene, the application is started with arguments that 
```

Comment:
Consider removing the first clause "Rather than..." so the sentence just says what the application does instead of comparing it another implicit design.